Source/monster.cpp

```diff
@@ -5018,4 +4991,55 @@ bool Monster::IsWalking() const
 	}
 }
 
+bool Monster::IsImmune(missile_id mName) const
+{
+	missile_resistance missileElement = MissilesData[mName].mResist;
+
+	if (((mMagicRes & IMMUNE_MAGIC) != 0 && missileElement == MISR_MAGIC)
+	    || ((mMagicRes & IMMUNE_FIRE) != 0 && missileElement == MISR_FIRE)
+	    || ((mMagicRes & IMMUNE_LIGHTNING) != 0 && missileElement == MISR_LIGHTNING)
+	    || ((mMagicRes & IMMUNE_ACID) != 0 && missileElement == MISR_ACID))
+		return true;
+	if (mName == MIS_HBOLT && MType->mtype != MT_DIABLO && MData->mMonstClass != MonsterClass::Undead)
+		return true;
+	return false;
+}
+
+bool Monster::IsResistant(missile_id mName) const
+{
+	missile_resistance missileElement = MissilesData[mName].mResist;
+
+	if (((mMagicRes & RESIST_MAGIC) != 0 && missileElement == MISR_MAGIC)
+	    || ((mMagicRes & RESIST_FIRE) != 0 && missileElement == MISR_FIRE)
+	    || ((mMagicRes & RESIST_LIGHTNING) != 0 && missileElement == MISR_LIGHTNING))
+		return true;
+	if (gbIsHellfire && mName == MIS_HBOLT && (MType->mtype == MT_DIABLO || MType->mtype == MT_BONEDEMN))
+		return true;
+	return false;
+}
+
+bool Monster::IsPossibleToHit() const
+{
+	if (_mhitpoints >> 6 <= 0
+	    || mtalkmsg != TEXT_NONE
+	    || (MType->mtype == MT_ILLWEAV && _mgoal == MGOAL_RETREAT)
+	    || _mmode == MonsterMode::Charge)
+		return false;
+	if ((MType->mtype >= MT_COUNSLR && MType->mtype <= MT_ADVOCATE) && _mgoal != MGOAL_NORMAL) {
+		GenerateRnd(100); // Vanilla Compatibility Note: Pushing RNG, as original function CheckMonsterHit was called after CTH roll

```

Comment:
No need to do so outside of DLRG or Item generation algorithms.

Reply:
Yeah, I know. But I am afraid it can a bit involve those chance to hit dirrerent outcome chains when player is cornering advocate (or even lazarus) and he is constantly trying to escape the corner by fading in and out. 
And I am afraid changing this tiny thing would make a bit different experience when player hits or misses this fading advo.
Leaving it here just assure game works as it did in original. Even if it is not the most intuitive way.
(But we also have firebolt damage recalculated every tick so :man_shrugging: )

Reply:
it wont make a perceptual difference, please remove it.